ft: ZENKO-1302 methods to add/remove ingestion readers

philipyoo opened the pull request:
**Purpose of this PR**:
Idea will be to use a separate ingestion reader for each new ingestion source.
The add/remove methods are to help instantiate new ingestion readers for each new ingestion source created (or removed). This PR only introduces the methods, and a later PR will use these methods along with the management client.
Also, in existing `QueuePopulator`, we avoid loading the ingestion extension to avoid overlap. Once things are more fleshed out for ingestion, maybe we can merge the processes. 

**Changes in this PR**:
- Refactor IngestionPopulator extension mgmt
- Refactor log messages (update with latest LogReader
  changes)
- Allow adding/removing IngestionReaders
- QueuePopulator should ignore ingestion extension. Separate
  out concerns

philipyoo commented:
@bert-e wait

bert-e commented:
# Hello philipyoo,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get
information on this process.





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @vrancurel

* @jonathan-gramain

* @rahulreddy

* @bennettbuchanan

philipyoo committed:
rf: ZENKO-1302 cleanup IngestionPopulator

Changes in this commit:
- Re-organize requires and variables at top of files
- Fix log messages to indicate ingestion process
- Remove unused failed consumer instantiation

philipyoo committed:
rf: ZENKO-1302 update logger in IngestionReader

Changes in this commit:
- Changes in PR#498 updates logger usage in LogReader.
  Since IngestionReader extends LogReader, this commit
  looks to update logger usage to match LogReader.

philipyoo committed:
ft: ZENKO-1302 separate crr and ingestion logs

Changes in this commit:
- Separate out concerns for QueuePopulator crr and
  IngestionPopulator logs

philipyoo committed:
ft: ZENKO-1302 allow adding/removing IngestionReaders

Changes in this commit:
- In preparation for workflow management, add methods to
  add or remove ingestion readers. This means that this
  change no longer supports the static config.json file
  source list and will rely on updates from Config.js

rahulreddy commented on lib/queuePopulator/IngestionPopulator.js:
```diff
@@ -131,66 +127,23 @@ class IngestionPopulator {
     _subscribeToLogSourceDispatcher(zkEndpoint, bucketd, ingestionPath) {
+        // NOTE: Bug where raftIdDispatcher will just be overwritten if more
+        // than one source exists. This will change in a later commit though
+        // when supporting workflow manager
```
Do you mean the `workflow manager`?

philipyoo replied:
sorry, I meant the initManagement process, I'll edit this

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers





Peer approvals *must* include at least 1 approval from the following list:

* @jonathan-gramain

* @bennettbuchanan

* @rahulreddy

* @vrancurel

rahulreddy commented on lib/queuePopulator/IngestionPopulator.js:
```diff
@@ -118,47 +116,46 @@ class IngestionPopulator {
             }
             const newRaftReaders = items.map(
                 token => {
+                    // TODO: will token ever be mongo?
```
I think Lifecycle moved away from using a dispatcher, so should we since there is only one process handling the ingestion from source. This token part can be removed as we will always ingest from a RaftLog.

philipyoo replied:
awesome! removing this part then

rahulreddy commented on lib/queuePopulator/IngestionProducer.js:
```diff
@@ -41,8 +41,7 @@ class IngestionProducer {
         this.s3source = s3Config;
-        this.bucketPrefix = sourceConfig.prefix ? sourceConfig.prefix :
-            sourceConfig.name;
+        this.zenkoName = sourceConfig.name;
```
~~What's a zenkoName?~~ Ah I see, zenko bucket name. We need a better name than that, like `targetZenkoBucket` or `zenkoBucket` to be descriptive.

rahulreddy reviewed (approved):
LGTM overall. It looks like we are headed in the right direction.

philipyoo commented:
@bert-e approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/8.1`



The following branches will **NOT** be impacted:

* `development/7.4`
* `development/8.0`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/8.1`



The following branches have **NOT** changed:

* `development/7.4`
* `development/8.0`



Please check the status of the associated issue ZENKO-1302.

Goodbye philipyoo.
